pkgs/ffigen/lib/src/code_generator/global.dart

```diff
@@ -41,35 +45,63 @@ class Global extends LookUpBinding {
     if (dartDoc != null) {
       s.write(makeDartDoc(dartDoc!));
     }
-    final pointerName = w.wrapperLevelUniqueNamer.makeUnique('_$globalVarName');
     final dartType = type.getFfiDartType(w);
     final cType = type.getCType(w);
 
-    s.write(
-        "late final ${w.ffiLibraryPrefix}.Pointer<$cType> $pointerName = ${w.lookupFuncIdentifier}<$cType>('$originalName');\n\n");
-    final baseTypealiasType = type.typealiasType;
-    if (baseTypealiasType is Compound) {
-      if (baseTypealiasType.isOpaque) {
-        s.write(
-            '${w.ffiLibraryPrefix}.Pointer<$cType> get $globalVarName => $pointerName;\n\n');
-      } else {
-        s.write('$dartType get $globalVarName => $pointerName.ref;\n\n');
+    if (nativeConfig.enabled) {
+      if (type case final ConstantArray arr) {
+        arr.generateSizeAnnotation(s, w);
+      }
+
+      s
+        ..writeln(makeNativeAnnotation(
+          w,
+          nativeType: cType,
+          assetId: nativeConfig.assetId,
+          differentName: globalVarName != originalName ? originalName : null,
+          isLeaf: false,
+        ))
+        ..write('external ');
+      if (constant) {
+        s.write('final ');
```

Comment:
Maybe I slightly prefer an `external get` than an `external final`. They are equivalent. But I feel getters are maybe slightly easier to understand for users?

Reply:
I have no strong feelings about the format, except that we should probably make this consistent. So either final/non-final fields or getters and a getter/setter pair for assignable fields. The latter would generate more code, but I can change it to that if we think it leads to more understandable outputs.

Reply:
Let's stick with the final fields then, we'll just have to educate our users. 👌